pkg/ocicni/ocicni_test.go

```diff
@@ -424,4 +424,41 @@ var _ = Describe("ocicni operations", func() {
 
 		ocicni.Shutdown()
 	})
+
+	It("update podCIDR", func() {
+		_, _, err := writeConfig(tmpDir, "10-test.conf", "test", "myplugin")
+		Expect(err).NotTo(HaveOccurred())
+
+		ocicni, err := InitCNI("test", tmpDir, "/opt/cni/bin")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ocicni.Status()).NotTo(HaveOccurred())
+
+		tmp := ocicni.(*cniNetworkPlugin)
+		Expect(tmp.podCidr).To(Equal(""))
+
+		podCidr := "192.168.0.0/24"
+		newPodCidr := "172.168.0.0/16"
+
+		// update pod cidr
+		ocicni.UpdateConfig(PodCidrUpdateType, podCidr)
+		Expect(tmp.podCidr).To(Equal(podCidr))
+
+		// try to update pod cidr to a new one
+		ocicni.UpdateConfig(PodCidrUpdateType, newPodCidr)
+		Expect(tmp.podCidr).To(Equal(podCidr))
+
+		ocicni.Shutdown()
+	})
+
+	It("buildCNIRuntimeConf with podCidr", func() {
+		podCidr := "192.168.0.0/24"
+
+		rt, err := buildCNIRuntimeConf("", &PodNetwork{}, "", "", podCidr)
+		Expect(err).NotTo(HaveOccurred())
+
+		ipRanges, ok := rt.CapabilityArgs["ipRanges"].([][]CniIpRange)
+		Expect(ok).To(Equal(true))
+		Expect(ipRanges[0][0].Subnet).To(Equal(podCidr))
+	})
+
```

Comment:
Does any of this test that an address from podCidr actually gets used in a container namespace?

Reply:
Sorry, only specific CNI plugin ([like a ptp plugin with podCidr capabilities enabled and using host-local as its IPAM plugin](https://github.com/containernetworking/plugins/blob/44297f6ba3dec024355f9a50d489ec0ed9335dd5/plugins/ipam/host-local/backend/allocator/config.go#L114)) can use this runtime config and get an valid address from podCidr, so I could not test using it but only can test passing it.